### PR TITLE
Apis integration

### PR DESCRIPTION
**Pull request by GorlemZ**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,12 +11,12 @@
   },
   "dependencies": {
     "@buildo/bento-design-system": "^0.20.4",
-    "i18next": "^23.5.1",
-    "prettier": "^3.0.3",
```
Comment (federico-ercoles) [inline package.json]: Were this intended to be removed? I think at least `i18next` is needed for `react-i18next`. I also don't see the dependency for `react-query`.

File: src/App.tsx [hunk 2]
```diff
@@ -1,14 +1,39 @@
...
 
 function App() {
   const { t } = useTranslation();
+  const { isLoading, isError, data } = useQuery({
+    queryKey: ["retrieve-list"],
+    queryFn: async () => {
+      const prom: JSON = await getRestaurantList(10);
+      return fromJsonToProp(prom);
+    },
+  });
```
Comment (federico-ercoles) [inline src/App.tsx]: Having this query defined here will result in it being executed also when a route different from `/home` is visited. Since there is no specific need to get the data here, I would move the query inside the `Home` component.

File: src/App.tsx [hunk 3]
```diff
@@ -1,14 +1,39 @@
...
+
+  if (isLoading) {
+    return <AreaLoader message="Loading..."></AreaLoader>;
+  }
+
+  if (isError) {
+    return <Toast message="Something went wrong" kind="negative"></Toast>;
+  }
```
Comment (federico-ercoles) [inline src/App.tsx]: Generally, UX-wise, since the data from the query is required to correctly display the page, I would consider displaying a `Feedback` component instead of showing a `Toast`.

File: src/apis/api.tsx [hunk 4]
```diff
@@ -0,0 +1,16 @@
+const apiKey = import.meta.env.VITE_API_KEY;
```
Comment (federico-ercoles) [inline src/apis/api.tsx]: Minor and not required: since you've added `zod`, consider setting up a runtime check of the env variables to ensure they are properly defined and structured.

File: src/components/RestaurantPreview.tsx [hunk 5]
```diff
@@ -1,25 +1,39 @@
...
       <Stack space={8}>
-        <img src={imageUrl} width="70%" height="100%" />
-        <Title size="medium">{name}</Title>
-        <Body size="medium">
-          {address}
+        <Box
+          height="full"
+          display="block"
+          justifyContent="center"
+          alignItems="center"
```
Comment (federico-ercoles) [inline src/components/RestaurantPreview.tsx]: With `display: block`, these two props should have no effect.

File: src/pages/Home.tsx [hunk 6]
```diff
@@ -40,11 +27,11 @@ function Home() {
...
       </Box>
 
-      <Inset space={16}>
+      <Inline collapseBelow={"desktop"} space={16}>
         <Tiles space={16} columns={3} alignY={"bottom"}>
-          {mockCards}
+          {cards}
         </Tiles>
-      </Inset>
+      </Inline>
```
Comment (federico-ercoles) [inline src/pages/Home.tsx]: Why this change from `Inset` to `Inline`? You're losing the padding around the grid. If you need to change the number of columns according to the viewport size, you can configure this in `Tiles`, e.g.:

```jsx
<Tiles space={16} columns={{ wide: 3, desktop: 3, tablet: 2, mobile: 1 }} alignY="bottom">
  {cards}
</Tiles>
```

File: src/App.tsx [hunk 7]
```diff
@@ -1,14 +1,39 @@
...
 import { useTranslation } from "react-i18next";
 import Home from "./pages/Home";
+import { useQuery } from "@tanstack/react-query";
+import { fromJsonToProp } from "./utils";
+import { getRestaurantList } from "./apis/api";
 
 function App() {
   const { t } = useTranslation();
+  const { isLoading, isError, data } = useQuery({
+    queryKey: ["retrieve-list"],
```
Comment (federico-ercoles) [inline src/App.tsx]: The query key should depend on the query parameters, so that queries with different parameters are cached separately and don't overwrite each other.

File: src/utils.tsx [hunk 8]
```diff
@@ -0,0 +1,10 @@
+import { PreviewList, previewList } from "./models";
+
+export function fromJsonToProp(jsonApi: JSON): PreviewList {
+  const propList = previewList.safeParse(jsonApi);
+  if (propList.success) {
+    return propList.data;
+  } else {
+    return { businesses: [] } as PreviewList;
```
Comment (federico-ercoles) [inline src/utils.tsx]: If this parsing fails, it should be handled as an error, instead of failing silently.

**Review by federico-ercoles — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 9]
```diff
@@ -8,6 +8,7 @@ pnpm-debug.log*
 lerna-debug.log*
 
 node_modules
+pnpm-lock*
```
Comment (federico-ercoles) [inline .gitignore]: Generally the lock-file should not be ignored.

File: src/pages/Home.tsx [hunk 10]
```diff
@@ -1,29 +1,44 @@
...
+import { getRestaurantList } from "../apis/api";
+import { fromJsonToProp } from "../utils";
+import { FiltersParams } from "../models";
 
-function Home() {
-  const name = rest_detail.name;
-  const imageUrl = rest_detail.image_url;
-  const rating = rest_detail.rating;
-  const address = rest_detail.location.address1;
+function Home({ range }: FiltersParams) {
```
Comment (federico-ercoles) [inline src/pages/Home.tsx]: I'm guessing you have this prop here because of my comment in my previous review. What I had in mind, though, was creating a custom hook `useGetRestaurantList` that wraps the `useQuery` usage, with a single `range` parameter.
Having a prop is useful when you need to pass data from outside the component, but here this doesn't make much sense.

File: src/models.tsx [hunk 11]
```diff
@@ -0,0 +1,34 @@
+import { z } from "zod";
+
+export const apisecret = z
```
Comment (federico-ercoles) [inline src/models.tsx]: Minor: missing camel-case.

File: src/pages/Home.tsx [hunk 12]
```diff
@@ -1,29 +1,44 @@
...
+      return fromJsonToProp(prom);
+    },
+  });
+
+  if (isLoading) {
+    return <AreaLoader message="Loading..."></AreaLoader>;
+  }
 
-  const mockCards = [...Array(16).keys()].map((element) => {
+  if (isError) {
```
Comment (federico-ercoles) [inline src/pages/Home.tsx:21]: Have you tested that, by throwing errors when the api call or the type checking fails, this `Feedback` is correctly displayed?

File: src/pages/Home.tsx (hunk 10 above)
Comment (GorlemZ) [reply src/pages/Home.tsx]: I'm quite confused, so: 
- where should i put/it is good practice to put the APIs calls? --> I'm thinking: eventually I'll need to use user-set params for requests. With these specific project requirements, everything could be in the Home component happily I guess, but what if the App component manage other pages that require APIs?
- "range" for now is a placeholder for all filters
- I got the "wrapper" issue, sounds right that the query should not be invoked with any kind of component refresh

File: src/pages/Home.tsx (hunk 12 above)
Comment (GorlemZ) [reply src/pages/Home.tsx:21]: yep (it is not centered, but works) //TBD

File: .gitignore (hunk 9 above)
Comment (GorlemZ) [reply .gitignore]: I'm mind blown by the fact that I could have solved my damn mess last week just running "pnpm install"

File: src/hooks.tsx [hunk 13]
```diff
@@ -0,0 +1,16 @@
...
+function useFetchQuery(range: number) {
+  const { isLoading, isError, data } = useQuery({
+    queryKey: ["retrieve-list", range],
+    queryFn: async () => {
+      const prom: JSON = await getRestaurantList(range);
+      return fromJsonToProp(prom);
+    },
+  });
+  return { data, isLoading, isError };
+}
```
Comment (federico-ercoles) [inline src/hooks.tsx]: A couple of minor suggestions:
* consider using a more explicit name for the hook, so that when you have multiples they are easily distinguishable. For example, it could be based on the name of the fetching function: `getRestaurantList -> useGetRestaurantList`.
* instead of extracting the specific fields you need now, consider returning the whole result of `useQuery`:
```suggestion
function useFetchQuery(range: number) {
  return useQuery({
    queryKey: ["retrieve-list", range],
    queryFn: async () => {
      const prom: JSON = await getRestaurantList(range);
      return fromJsonToProp(prom);
    },
  });
}
```

**Review by federico-ercoles — COMMENTED** [review]
Looks 💯 ! Just a couple of minor improvement suggestions.

**Review by federico-ercoles — APPROVED** [review]
🚀
